### ISSUE TITLE
memory-cache is a memory leak?

### ISSUE THREAD

**Issue by goofballLogic** [ForbesLindesay/http-basic#19]
Unless I misunderstand, it appears that "memory" cached responses are never automatically invalidated. This means that, unless you invalidate responses through a PUT or POST etc., the memory footprint will grow and grow. Obviously, this isn't practical unless you have a strictly limited number of URLs which will be called.

**Comment by ForbesLindesay** [comment]
Yes, the builtin caches are good enough for my use cases (mostly short running CLI scripts etc.) but if you need http caching in a production environment, you should implement the cache interface.  See https://github.com/ForbesLindesay/http-basic#implementing-a-cache

If you have the time to improve the builtin caches, or add extra implementations (e.g. it would be cool to have a redis & a MongoDB cache) I would be happy to accept pull requests.
